### ISSUE TITLE
hostnames <table> in greeting banner does not consider ipv6

### ISSUE THREAD

**Issue by pywy18** [OpenSMTPD/OpenSMTPD#1067]
Using opensmtpd on OpenBSD 6.7 stable here
 
In a configuration using a table for greeting banner, the ipv6 adresses are not considered : 
`listen on em0 port 25 hostnames <myhostnames>`
with the table content : 
```
192.168.0.1   my.custom.domain.tld
2001:db8:0::1  another.custom.domain.tld
```
Whenever I connect on the ipv4 addr, the right domain shows up, but the default domain is displayed when I connect using ipv6, in stead of the one specified in table.

**Comment by poolpOrg** [comment]
Issue reproduced and understood, will work on a fix.

Basically, the problem is that for multi-column tables, the table parser in table_static.c uses the first ':' as a field separator since it is the delimiter for aliases. In this cases, this results in:

```
2001:db8:0::1  another.custom.domain.tld
```

being understood as:

```
2001 = db8:0::1  another.custom.domain.tld
```

**Comment by poolpOrg** [comment]
Will let upstream know about this issue so they can fix

**Comment by innogrey** [comment]
Was this issue ever resolved? It seems that 6.8.0p2 still doesn't allow putting v6 addresses in a table.

**Comment by SWW13** [comment]
It looks like it's working at least for some tables now with 7.8.0p0.

I was able to use it in a `helo-src <helo>` table like this:
```
[2001:db8:0::1]  another.custom.domain.tld
```

**Comment by omar-polo** [comment]
Yeah, I think I've fixed this in https://github.com/OpenSMTPD/OpenSMTPD/commit/9b9a387cc348b0e8178bd85100129b198e793c46 back in 7.6, but I didn't know there was an issue tracking this and so haven't updated.  apologize :)

I'm going to close this for now, feel free to reopen it in case.

Thanks!
